Std/Data/Bool.lean

```diff
@@ -118,31 +128,31 @@ alias not_or := Bool.or_deMorgan
 
 /-! ### xor -/
 
-@[simp] theorem xor_false_left : ∀ (x : Bool), xor false x = x := by
+@[simp] theorem false_xor : ∀ (x : Bool), xor false x = x := by
   decide
 
-@[simp] theorem xor_false_right : ∀ (x : Bool), xor x false = x := by
+@[simp] theorem xor_false : ∀ (x : Bool), xor x false = x := by
   decide
 
-@[simp] theorem xor_true_left : ∀ (x : Bool), xor true x = !x := by
+@[simp] theorem true_xor : ∀ (x : Bool), xor true x = !x := by
   decide
 
-@[simp] theorem xor_true_right : ∀ (x : Bool), xor x true = !x := by
+@[simp] theorem xor_true : ∀ (x : Bool), xor x true = !x := by
   decide
 
-@[simp] theorem xor_not_self_left : ∀ (x : Bool), xor (!x) x = true := by
+@[simp] theorem not_xor_self : ∀ (x : Bool), xor (!x) x = true := by
   decide
 
-@[simp] theorem xor_not_self_right : ∀ (x : Bool), xor x (!x) = true := by
+@[simp] theorem xor_not_self : ∀ (x : Bool), xor x (!x) = true := by
   decide
 
-@[simp] theorem xor_not_left : ∀ (x y : Bool), xor (!x) y = !(xor x y) := by
+theorem not_xor : ∀ (x y : Bool), xor (!x) y = !(xor x y) := by
   decide
 
-@[simp] theorem xor_not_right : ∀ (x y : Bool), xor x (!y) = !(xor x y) := by
+theorem xor_not : ∀ (x y : Bool), xor x (!y) = !(xor x y) := by
   decide
 
-theorem xor_not_not : ∀ (x y : Bool), xor (!x) (!y) = (xor x y) := by
+theorem not_xor_not : ∀ (x y : Bool), xor (!x) (!y) = (xor x y) := by
   decide
```

Comment:
Is the double negation rule here used elsewhere?

It seems like the two above theorems (plus cancelling not) would remove its need.

Reply:
Good point. It's here because it `not_xor_not` exists in Mathlib and Mathlib's versions of `not_xor` and `xor_not` conflate the two arguments into one. This might be a mistake on Mathlib's end, I never figured out why it was that way.

Anyway, I'm on the fence here because (even though I just made them simp in my last commit!) it's not entirely clear to me that `not_xor` and `xor_not` should be simp, but `not_xor_not` is definitely a simp candidate. (I tend to be very conservative about simp so this is perhaps not a useful distinction.)

I'm happy to remove `not_xor_not` if that's the consensus.